### ISSUE TITLE
PHP-FPM 7.3 possibly breaks fog web interface on Arch

### ISSUE THREAD

**Issue by ase1590** [FOGProject/fogproject#281]
**OS: Arch Linux**
Kernel: 4.20

FOG version: 1.5.5 & 'working-1.6'

Packages:
apache: 2.4.37-1
mariadb: 10.1.37-1
php-fpm: 7.3.0-1
php-apache: 7.3.0-1
php: 7.3.0.1
php-gd: 7.3.0-1

attempting to go to the FOG dashboard results in 'Service unavailable! Error 503'

this started happening right around when PHP 7.3 landed. 

combined /opt/fog logs: http://ix.io/1xKb
httpd error_log: http://ix.io/1xKd
`systemctl status php-fpm`: http://ix.io/1xKg

`apachectl -t` returns `syntax ok`

apache is using `mpm_prefork_module`

this happened originally on 1.5.5, but I also tested this on the working 1.6 branch as well. same results.

**Comment by mastacontrola** [comment]
I don't know what's causing the error with php-fpm on 7.3. I have adjusted all the regex commands to escape hyphens where we use them as a part of the string search, but this did not help anything. There appears to be a problem with the fpm implementation and I don't have the time right now to figure it all out. To correct for this, I've added a minimum and maximum versioning for PHP. For right now, I know php 7.2 works perfectly with FOG on both 1.5.5 and 1.6 and I've added the comparison checking update to the working-1.6.

Please downgrade your php version to the latest 7.2 and you should be fine.

**Comment by ase1590** [comment]
Will do. glad its not just me then :)

**Comment by Sebastian-Roth** [comment]
@ase1590 Thanks for bringing this up. I have started to look into this at the end of 2018 but couldn't actually figure out why it fails. But as we didn't have any user reporting this so far I did not put too much time into it.

I will give it another try and see what I come up with...

**Comment by mastacontrola** [comment]
So I just tested today with installing PHP version 7.3.1 and all seems to be working now. Maybe there was a bug in the php-fpm implementation that PHP initially released as a part of 7.3 that's now been corrected?

**Comment by mastacontrola** [comment]
```
 * Starting FOGImageReplicator.service Service.................OK
 * Starting FOGSnapinReplicator.service Service................OK
 * Starting FOGScheduler.service Service.......................OK
 * Starting FOGPingHosts.service Service.......................OK
 * Starting FOGSnapinHash.service Service......................OK
 * Starting FOGImageSize.service Service.......................OK
 * Setting up exports file.....................................OK
 * Setting up and starting RPCBind.............................OK
 * Setting up and starting NFS Server..........................OK
 * Ensuring node username and passwords match..................Done

 * Setup complete

   You can now login to the FOG Management Portal using
   the information listed below.  The login information
   is only if this is the first install.

   This can be done by opening a web browser and going to:

   http://192.168.56.15/fog/management

   Default User Information
   Username: fog
   Password: password

[root@fogstorage bin]# php -v
PHP 7.3.1 (cli) (built: Jan  8 2019 13:55:51) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.1, Copyright (c) 1998-2018 Zend Technologies
```

**Comment by Sebastian-Roth** [comment]
Yeah, definitely right. Seems to be fixed now with PHP version 7.3.1! Still there seems to be an issue in the arch installation as `mysql_install_db` is not urn properly. So if you have a DB issue after the installer finished please run `mysql_install_db --user=mysql --basedir=/usr --datadir=/var/lib/mysql` and `systemctl start mariadb` as root and you should be all set. I just fixed the installer in the `dev-branch`.

I am closing this issue as the initial problem is solved - tested several times.

**Comment by ase1590** [comment]
Yeah seems to be running fine now with 7.3.1!
